pkg/kapp/cmd/core/config_factory.go

```diff
@@ -113,7 +113,12 @@ func (f *ConfigFactoryImpl) clientConfig() (bool, clientcmd.ClientConfig, error)
 	}
 
 	if len(configYAML) > 0 {
-		envHostPort := net.JoinHostPort(os.Getenv("KUBERNETES_SERVICE_HOST"), os.Getenv("KUBERNETES_SERVICE_PORT"))
+		// If it is not provided, we default the Kubernetes service port to 443
+		kubernetesServicePort, ok := os.LookupEnv("KUBERNETES_SERVICE_PORT")
+		if !ok {
+			kubernetesServicePort = "443"
+		}
+		envHostPort := net.JoinHostPort(os.Getenv("KUBERNETES_SERVICE_HOST"), kubernetesServicePort)
```

Comment:
Instead of manually adding the port when it is empty, let's not use the `JoinHostPort` (which adds the `:` at the end) and directly use the host. client-go will manually add the port based on `http`/`https`.

Reply:
Sure!